planar/area.go

```diff
@@ -190,13 +190,13 @@ func polygonCentroidArea(p orb.Polygon) (orb.Point, float64) {
 	area = math.Abs(area)
 
 	holeArea := 0.0
-	holeCentroid := orb.Point{}
+	weightedHoleCentroid := orb.Point{}
 	for i := 1; i < len(p); i++ {
 		hc, ha := ringCentroidArea(p[i])
 
 		holeArea += math.Abs(ha)
-		holeCentroid[0] += hc[0] * ha
-		holeCentroid[1] += hc[1] * ha
+		weightedHoleCentroid[0] += hc[0] * ha
```

Comment:
actually I think you need something like:
```
hc, ha := ringCentroidArea(p[i])
ha = math.Abs(ha)

holeArea += ha
weightedHoleCentroid[0] += hc[0] * ha
weightedHoleCentroid[1] += hc[1] * ha
```


I wrote this test that still failed with your changes:
```
	// empty right half
	poly = orb.Polygon{
		{{0, 0}, {4, 0}, {4, 4}, {0, 4}, {0, 0}},
		{{2, 0}, {2, 4}, {4, 4}, {4, 0}, {2, 0}},
	}

	centroid, area = CentroidArea(poly)
	if v := (orb.Point{1, 2}); !centroid.Equal(v) {
		t.Errorf("incorrect centroid: %v != %v", centroid, v)
	}

	if area != 8 {
		t.Errorf("incorrect area: %v != 8", area)
	}
```